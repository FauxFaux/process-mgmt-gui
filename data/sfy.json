{
  "items": {
    "adaptive_control_unit": {
      "name": "Adaptive Control Unit",
      "labId": "adaptive-control-unit",
      "stack": 50,
      "iconPos": "-0px -0px"
    },
    "ai_limiter": {
      "name": "AI Limiter",
      "labId": "ai-limiter",
      "stack": 100,
      "iconPos": "-64px -0px"
    },
    "alclad_aluminum_sheet": {
      "name": "Alclad Aluminum Sheet",
      "labId": "alclad-aluminum-sheet",
      "stack": 200,
      "iconPos": "-128px -0px"
    },
    "alien_dna_capsule": {
      "name": "Alien DNA Capsule",
      "labId": "alien-dna-capsule",
      "stack": 50,
      "iconPos": "-192px -0px"
    },
    "alien_protein": {
      "name": "Alien Protein",
      "labId": "alien-protein",
      "stack": 100,
      "iconPos": "-256px -0px"
    },
    "alumina_solution": {
      "name": "Alumina Solution",
      "labId": "alumina-solution",
      "iconPos": "-320px -0px"
    },
    "aluminum_casing": {
      "name": "Aluminum Casing",
      "labId": "aluminum-casing",
      "stack": 200,
      "iconPos": "-384px -0px"
    },
    "aluminum_ingot": {
      "name": "Aluminum Ingot",
      "labId": "aluminum-ingot",
      "stack": 100,
      "iconPos": "-448px -0px"
    },
    "aluminum_scrap": {
      "name": "Aluminum Scrap",
      "labId": "aluminum-scrap",
      "stack": 500,
      "iconPos": "-512px -0px"
    },
    "assembler": {
      "name": "Assembler",
      "iconPos": "-576px -0px"
    },
    "assembly_director_system": {
      "name": "Assembly Director System",
      "labId": "assembly-director-system",
      "stack": 50,
      "iconPos": "-640px -0px"
    },
    "automated_wiring": {
      "name": "Automated Wiring",
      "labId": "automated-wiring",
      "stack": 50,
      "iconPos": "-704px -0px"
    },
    "battery": {
      "name": "Battery",
      "stack": 200,
      "iconPos": "-768px -0px"
    },
    "bauxite": {
      "name": "Bauxite",
      "stack": 100,
      "iconPos": "-0px -64px"
    },
    "beacon": {
      "name": "Beacon",
      "stack": 100,
      "iconPos": "-64px -64px"
    },
    "biomass": {
      "name": "Biomass",
      "stack": 200,
      "iconPos": "-192px -64px"
    },
    "black_powder": {
      "name": "Black Powder",
      "labId": "black-powder",
      "stack": 200,
      "iconPos": "-256px -64px"
    },
    "blender": {
      "name": "Blender",
      "iconPos": "-320px -64px"
    },
    "blue_power_slug": {
      "name": "Blue Power Slug",
      "labId": "blue-power-slug",
      "stack": 50,
      "iconPos": "-384px -64px"
    },
    "cable": {
      "name": "Cable",
      "stack": 200,
      "iconPos": "-448px -64px"
    },
    "caterium_ingot": {
      "name": "Caterium Ingot",
      "labId": "caterium-ingot",
      "stack": 100,
      "iconPos": "-512px -64px"
    },
    "caterium_ore": {
      "name": "Caterium Ore",
      "labId": "caterium-ore",
      "stack": 100,
      "iconPos": "-576px -64px"
    },
    "circuit_board": {
      "name": "Circuit Board",
      "labId": "circuit-board",
      "stack": 200,
      "iconPos": "-640px -64px"
    },
    "cluster_nobelisk": {
      "name": "Cluster Nobelisk",
      "labId": "cluster-nobelisk",
      "stack": 50,
      "iconPos": "-704px -64px"
    },
    "coal": {
      "name": "Coal",
      "stack": 100,
      "iconPos": "-0px -128px"
    },
    "color_cartridge": {
      "name": "Color Cartridge",
      "labId": "color-cartridge",
      "stack": 200,
      "iconPos": "-64px -128px"
    },
    "compacted_coal": {
      "name": "Compacted Coal",
      "labId": "compacted-coal",
      "stack": 100,
      "iconPos": "-128px -128px"
    },
    "computer": {
      "name": "Computer",
      "stack": 50,
      "iconPos": "-192px -128px"
    },
    "concrete": {
      "name": "Concrete",
      "stack": 500,
      "iconPos": "-256px -128px"
    },
    "constructor_id": {
      "name": "Constructor",
      "labId": "constructor-id",
      "iconPos": "-320px -128px"
    },
    "conveyor_belt_mk1": {
      "name": "Conveyor Belt Mk.1",
      "labId": "conveyor-belt-mk1",
      "iconPos": "-384px -128px"
    },
    "conveyor_belt_mk2": {
      "name": "Conveyor Belt Mk.2",
      "labId": "conveyor-belt-mk2",
      "stack": 1,
      "iconPos": "-448px -128px"
    },
    "conveyor_belt_mk3": {
      "name": "Conveyor Belt Mk.3",
      "labId": "conveyor-belt-mk3",
      "iconPos": "-512px -128px"
    },
    "conveyor_belt_mk4": {
      "name": "Conveyor Belt Mk.4",
      "labId": "conveyor-belt-mk4",
      "iconPos": "-576px -128px"
    },
    "conveyor_belt_mk5": {
      "name": "Conveyor Belt Mk.5",
      "labId": "conveyor-belt-mk5",
      "iconPos": "-640px -128px"
    },
    "cooling_system": {
      "name": "Cooling System",
      "labId": "cooling-system",
      "stack": 100,
      "iconPos": "-704px -128px"
    },
    "copper_ingot": {
      "name": "Copper Ingot",
      "labId": "copper-ingot",
      "stack": 100,
      "iconPos": "-768px -128px"
    },
    "copper_ore": {
      "name": "Copper Ore",
      "labId": "copper-ore",
      "stack": 100,
      "iconPos": "-0px -192px"
    },
    "copper_powder": {
      "name": "Copper Powder",
      "labId": "copper-powder",
      "stack": 500,
      "iconPos": "-64px -192px"
    },
    "copper_sheet": {
      "name": "Copper Sheet",
      "labId": "copper-sheet",
      "stack": 200,
      "iconPos": "-128px -192px"
    },
    "crude_oil": {
      "name": "Crude Oil",
      "labId": "crude-oil",
      "iconPos": "-192px -192px"
    },
    "crystal_oscillator": {
      "name": "Crystal Oscillator",
      "labId": "crystal-oscillator",
      "stack": 100,
      "iconPos": "-256px -192px"
    },
    "electromagnetic_control_rod": {
      "name": "Electromagnetic Control Rod",
      "labId": "electromagnetic-control-rod",
      "stack": 100,
      "iconPos": "-320px -192px"
    },
    "empty_canister": {
      "name": "Empty Canister",
      "labId": "empty-canister",
      "stack": 100,
      "iconPos": "-384px -192px"
    },
    "empty_fluid_tank": {
      "name": "Empty Fluid Tank",
      "labId": "empty-fluid-tank",
      "stack": 100,
      "iconPos": "-448px -192px"
    },
    "encased_industrial_beam": {
      "name": "Encased Industrial Beam",
      "labId": "encased-industrial-beam",
      "stack": 100,
      "iconPos": "-512px -192px"
    },
    "encased_plutonium_cell": {
      "name": "Encased Plutonium Cell",
      "labId": "encased-plutonium-cell",
      "stack": 200,
      "iconPos": "-576px -192px"
    },
    "encased_uranium_cell": {
      "name": "Encased Uranium Cell",
      "labId": "encased-uranium-cell",
      "stack": 200,
      "iconPos": "-640px -192px"
    },
    "explosive_rebar": {
      "name": "Explosive Rebar",
      "labId": "explosive-rebar",
      "stack": 50,
      "iconPos": "-704px -192px"
    },
    "fabric": {
      "name": "Fabric",
      "stack": 100,
      "iconPos": "-768px -192px"
    },
    "flower_petals": {
      "name": "Flower Petals",
      "labId": "flower-petals",
      "stack": 500,
      "iconPos": "-0px -256px"
    },
    "foundry": {
      "name": "Foundry",
      "iconPos": "-64px -256px"
    },
    "freight_car": {
      "name": "Freight Car",
      "labId": "freight-car",
      "iconPos": "-128px -256px"
    },
    "fuel": {
      "name": "Fuel",
      "iconPos": "-256px -256px"
    },
    "fused_modular_frame": {
      "name": "Fused Modular Frame",
      "labId": "fused-modular-frame",
      "stack": 50,
      "iconPos": "-320px -256px"
    },
    "gas_filter": {
      "name": "Gas Filter",
      "labId": "gas-filter",
      "stack": 50,
      "iconPos": "-384px -256px"
    },
    "gas_nobelisk": {
      "name": "Gas Nobelisk",
      "labId": "gas-nobelisk",
      "stack": 50,
      "iconPos": "-448px -256px"
    },
    "hatcher_remains": {
      "name": "Hatcher Remains",
      "labId": "hatcher-remains",
      "stack": 100,
      "iconPos": "-576px -256px"
    },
    "heat_sink": {
      "name": "Heat Sink",
      "labId": "heat-sink",
      "stack": 100,
      "iconPos": "-640px -256px"
    },
    "heavy_modular_frame": {
      "name": "Heavy Modular Frame",
      "labId": "heavy-modular-frame",
      "stack": 50,
      "iconPos": "-704px -256px"
    },
    "heavy_oil_residue": {
      "name": "Heavy Oil Residue",
      "labId": "heavy-oil-residue",
      "iconPos": "-768px -256px"
    },
    "high_speed_connector": {
      "name": "High-Speed Connector",
      "labId": "high-speed-connector",
      "stack": 100,
      "iconPos": "-0px -320px"
    },
    "hog_remains": {
      "name": "Hog Remains",
      "labId": "hog-remains",
      "stack": 100,
      "iconPos": "-64px -320px"
    },
    "homing_rifle_ammo": {
      "name": "Homing Rifle Ammo",
      "labId": "homing-rifle-ammo",
      "stack": 100,
      "iconPos": "-128px -320px"
    },
    "iodine_infused_filter": {
      "name": "Iodine Infused Filter",
      "labId": "iodine-infused-filter",
      "stack": 50,
      "iconPos": "-192px -320px"
    },
    "iron_ingot": {
      "name": "Iron Ingot",
      "labId": "iron-ingot",
      "stack": 100,
      "iconPos": "-256px -320px"
    },
    "iron_ore": {
      "name": "Iron Ore",
      "labId": "iron-ore",
      "stack": 100,
      "iconPos": "-320px -320px"
    },
    "iron_plate": {
      "name": "Iron Plate",
      "labId": "iron-plate",
      "stack": 200,
      "iconPos": "-384px -320px"
    },
    "iron_rebar": {
      "name": "Iron Rebar",
      "labId": "iron-rebar",
      "stack": 50,
      "iconPos": "-448px -320px"
    },
    "iron_rod": {
      "name": "Iron Rod",
      "labId": "iron-rod",
      "stack": 200,
      "iconPos": "-512px -320px"
    },
    "leaves": {
      "name": "Leaves",
      "stack": 500,
      "iconPos": "-576px -320px"
    },
    "limestone": {
      "name": "Limestone",
      "stack": 100,
      "iconPos": "-640px -320px"
    },
    "liquid_biofuel": {
      "name": "Liquid Biofuel",
      "labId": "liquid-biofuel",
      "iconPos": "-704px -320px"
    },
    "magnetic_field_generator": {
      "name": "Magnetic Field Generator",
      "labId": "magnetic-field-generator",
      "stack": 50,
      "iconPos": "-768px -320px"
    },
    "manufacturer": {
      "name": "Manufacturer",
      "iconPos": "-0px -384px"
    },
    "miner_mk1": {
      "name": "Miner Mk.1",
      "labId": "miner-mk1",
      "iconPos": "-64px -384px"
    },
    "miner_mk2": {
      "name": "Miner Mk.2",
      "labId": "miner-mk2",
      "iconPos": "-128px -384px"
    },
    "miner_mk3": {
      "name": "Miner Mk.3",
      "labId": "miner-mk3",
      "iconPos": "-192px -384px"
    },
    "modular_engine": {
      "name": "Modular Engine",
      "labId": "modular-engine",
      "stack": 50,
      "iconPos": "-256px -384px"
    },
    "modular_frame": {
      "name": "Modular Frame",
      "labId": "modular-frame",
      "stack": 50,
      "iconPos": "-320px -384px"
    },
    "motor": {
      "name": "Motor",
      "stack": 50,
      "iconPos": "-384px -384px"
    },
    "mycelia": {
      "name": "Mycelia",
      "stack": 200,
      "iconPos": "-448px -384px"
    },
    "nitric_acid": {
      "name": "Nitric Acid",
      "labId": "nitric-acid",
      "iconPos": "-512px -384px"
    },
    "nitrogen_gas": {
      "name": "Nitrogen Gas",
      "labId": "nitrogen-gas",
      "iconPos": "-576px -384px"
    },
    "nobelisk": {
      "name": "Nobelisk",
      "stack": 50,
      "iconPos": "-640px -384px"
    },
    "non_fissile_uranium": {
      "name": "Non-fissile Uranium",
      "labId": "non-fissile-uranium",
      "stack": 500,
      "iconPos": "-704px -384px"
    },
    "nuclear_pasta": {
      "name": "Nuclear Pasta",
      "labId": "nuclear-pasta",
      "stack": 50,
      "iconPos": "-768px -384px"
    },
    "nuclear_power_plant": {
      "name": "Nuclear Power Plant",
      "labId": "nuclear-power-plant",
      "iconPos": "-0px -448px"
    },
    "nuke_nobelisk": {
      "name": "Nuke Nobelisk",
      "labId": "nuke-nobelisk",
      "stack": 50,
      "iconPos": "-64px -448px"
    },
    "oil_extractor": {
      "name": "Oil Extractor",
      "labId": "oil-extractor",
      "iconPos": "-128px -448px"
    },
    "packaged_alumina_solution": {
      "name": "Packaged Alumina Solution",
      "labId": "packaged-alumina-solution",
      "stack": 100,
      "iconPos": "-192px -448px"
    },
    "packaged_fuel": {
      "name": "Packaged Fuel",
      "labId": "packaged-fuel",
      "stack": 100,
      "iconPos": "-256px -448px"
    },
    "packaged_heavy_oil_residue": {
      "name": "Packaged Heavy Oil Residue",
      "labId": "packaged-heavy-oil-residue",
      "stack": 100,
      "iconPos": "-320px -448px"
    },
    "packaged_liquid_biofuel": {
      "name": "Packaged Liquid Biofuel",
      "labId": "packaged-liquid-biofuel",
      "stack": 100,
      "iconPos": "-384px -448px"
    },
    "packaged_nitric_acid": {
      "name": "Packaged Nitric Acid",
      "labId": "packaged-nitric-acid",
      "stack": 100,
      "iconPos": "-448px -448px"
    },
    "packaged_nitrogen_gas": {
      "name": "Packaged Nitrogen Gas",
      "labId": "packaged-nitrogen-gas",
      "stack": 100,
      "iconPos": "-512px -448px"
    },
    "packaged_oil": {
      "name": "Packaged Oil",
      "labId": "packaged-oil",
      "stack": 100,
      "iconPos": "-576px -448px"
    },
    "packaged_sulfuric_acid": {
      "name": "Packaged Sulfuric Acid",
      "labId": "packaged-sulfuric-acid",
      "stack": 100,
      "iconPos": "-640px -448px"
    },
    "packaged_turbofuel": {
      "name": "Packaged Turbofuel",
      "labId": "packaged-turbofuel",
      "stack": 100,
      "iconPos": "-704px -448px"
    },
    "packaged_water": {
      "name": "Packaged Water",
      "labId": "packaged-water",
      "stack": 100,
      "iconPos": "-768px -448px"
    },
    "packager": {
      "name": "Packager",
      "iconPos": "-0px -512px"
    },
    "particle_accelerator": {
      "name": "Particle Accelerator",
      "labId": "particle-accelerator",
      "iconPos": "-64px -512px"
    },
    "petroleum_coke": {
      "name": "Petroleum Coke",
      "labId": "petroleum-coke",
      "stack": 200,
      "iconPos": "-128px -512px"
    },
    "pipeline_mk1": {
      "name": "Pipeline Mk.1",
      "labId": "pipeline-mk1",
      "iconPos": "-192px -512px"
    },
    "pipeline_mk2": {
      "name": "Pipeline Mk.2",
      "labId": "pipeline-mk2",
      "iconPos": "-256px -512px"
    },
    "plastic": {
      "name": "Plastic",
      "stack": 200,
      "iconPos": "-320px -512px"
    },
    "plutonium_fuel_rod": {
      "name": "Plutonium Fuel Rod",
      "labId": "plutonium-fuel-rod",
      "stack": 50,
      "iconPos": "-384px -512px"
    },
    "plutonium_pellet": {
      "name": "Plutonium Pellet",
      "labId": "plutonium-pellet",
      "stack": 100,
      "iconPos": "-448px -512px"
    },
    "plutonium_waste": {
      "name": "Plutonium Waste",
      "labId": "plutonium-waste",
      "stack": 500,
      "iconPos": "-512px -512px"
    },
    "polymer_resin": {
      "name": "Polymer Resin",
      "labId": "polymer-resin",
      "stack": 200,
      "iconPos": "-576px -512px"
    },
    "portable_miner": {
      "name": "Portable Miner",
      "labId": "portable-miner",
      "stack": 1,
      "iconPos": "-640px -512px"
    },
    "power_shard": {
      "name": "Power Shard",
      "labId": "power-shard",
      "stack": 100,
      "iconPos": "-704px -512px"
    },
    "pressure_conversion_cube": {
      "name": "Pressure Conversion Cube",
      "labId": "pressure-conversion-cube",
      "stack": 50,
      "iconPos": "-0px -576px"
    },
    "pulse_nobelisk": {
      "name": "Pulse Nobelisk",
      "labId": "pulse-nobelisk",
      "stack": 50,
      "iconPos": "-64px -576px"
    },
    "purity_0": {
      "name": "Impure",
      "labId": "purity-0"
    },
    "purity_1": {
      "name": "Normal",
      "labId": "purity-1"
    },
    "purity_2": {
      "name": "Pure",
      "labId": "purity-2"
    },
    "purple_power_slug": {
      "name": "Purple Power Slug",
      "labId": "purple-power-slug",
      "stack": 50,
      "iconPos": "-128px -576px"
    },
    "quartz_crystal": {
      "name": "Quartz Crystal",
      "labId": "quartz-crystal",
      "stack": 200,
      "iconPos": "-192px -576px"
    },
    "quickwire": {
      "name": "Quickwire",
      "stack": 500,
      "iconPos": "-256px -576px"
    },
    "radio_control_unit": {
      "name": "Radio Control Unit",
      "labId": "radio-control-unit",
      "stack": 50,
      "iconPos": "-320px -576px"
    },
    "raw_quartz": {
      "name": "Raw Quartz",
      "labId": "raw-quartz",
      "stack": 100,
      "iconPos": "-384px -576px"
    },
    "refinery": {
      "name": "Refinery",
      "iconPos": "-448px -576px"
    },
    "reinforced_iron_plate": {
      "name": "Reinforced Iron Plate",
      "labId": "reinforced-iron-plate",
      "stack": 100,
      "iconPos": "-512px -576px"
    },
    "resource_well_extractor": {
      "name": "Resource Well Extractor",
      "labId": "resource-well-extractor",
      "iconPos": "-576px -576px"
    },
    "resource_well_pressurizer": {
      "name": "Resource Well Pressurizer",
      "labId": "resource-well-pressurizer",
      "iconPos": "-640px -576px"
    },
    "rifle_cartridge": {
      "name": "Rifle Ammo",
      "labId": "rifle-cartridge",
      "stack": 500,
      "iconPos": "-704px -576px"
    },
    "rotor": {
      "name": "Rotor",
      "stack": 100,
      "iconPos": "-768px -576px"
    },
    "rubber": {
      "name": "Rubber",
      "stack": 200,
      "iconPos": "-0px -640px"
    },
    "sam_ore": {
      "name": "SAM Ore",
      "labId": "sam-ore",
      "stack": 50,
      "iconPos": "-64px -640px"
    },
    "screw": {
      "name": "Screw",
      "stack": 500,
      "iconPos": "-128px -640px"
    },
    "shatter_rebar": {
      "name": "Shatter Rebar",
      "labId": "shatter-rebar",
      "stack": 50,
      "iconPos": "-192px -640px"
    },
    "silica": {
      "name": "Silica",
      "stack": 100,
      "iconPos": "-256px -640px"
    },
    "smart_plating": {
      "name": "Smart Plating",
      "labId": "smart-plating",
      "stack": 50,
      "iconPos": "-320px -640px"
    },
    "smelter": {
      "name": "Smelter",
      "iconPos": "-384px -640px"
    },
    "smokeless_powder": {
      "name": "Smokeless Powder",
      "labId": "smokeless-powder",
      "stack": 100,
      "iconPos": "-448px -640px"
    },
    "solid_biofuel": {
      "name": "Solid Biofuel",
      "labId": "solid-biofuel",
      "stack": 200,
      "iconPos": "-512px -640px"
    },
    "spitter_remains": {
      "name": "Spitter Remains",
      "labId": "spitter-remains",
      "stack": 100,
      "iconPos": "-576px -640px"
    },
    "stator": {
      "name": "Stator",
      "stack": 100,
      "iconPos": "-640px -640px"
    },
    "steel_beam": {
      "name": "Steel Beam",
      "labId": "steel-beam",
      "stack": 200,
      "iconPos": "-704px -640px"
    },
    "steel_ingot": {
      "name": "Steel Ingot",
      "labId": "steel-ingot",
      "stack": 100,
      "iconPos": "-768px -640px"
    },
    "steel_pipe": {
      "name": "Steel Pipe",
      "labId": "steel-pipe",
      "stack": 200,
      "iconPos": "-0px -704px"
    },
    "stinger_remains": {
      "name": "Stinger Remains",
      "labId": "stinger-remains",
      "stack": 100,
      "iconPos": "-64px -704px"
    },
    "stun_rebar": {
      "name": "Stun Rebar",
      "labId": "stun-rebar",
      "stack": 50,
      "iconPos": "-128px -704px"
    },
    "sulfur": {
      "name": "Sulfur",
      "stack": 100,
      "iconPos": "-192px -704px"
    },
    "sulfuric_acid": {
      "name": "Sulfuric Acid",
      "labId": "sulfuric-acid",
      "iconPos": "-256px -704px"
    },
    "supercomputer": {
      "name": "Supercomputer",
      "stack": 50,
      "iconPos": "-320px -704px"
    },
    "thermal_propulsion_rocket": {
      "name": "Thermal Propulsion Rocket",
      "labId": "thermal-propulsion-rocket",
      "stack": 50,
      "iconPos": "-384px -704px"
    },
    "turbo_motor": {
      "name": "Turbo Motor",
      "labId": "turbo-motor",
      "stack": 50,
      "iconPos": "-448px -704px"
    },
    "turbo_rifle_ammo": {
      "name": "Turbo Rifle Ammo",
      "labId": "turbo-rifle-ammo",
      "stack": 100,
      "iconPos": "-512px -704px"
    },
    "turbofuel": {
      "name": "Turbofuel",
      "iconPos": "-576px -704px"
    },
    "uranium": {
      "name": "Uranium",
      "stack": 100,
      "iconPos": "-768px -704px"
    },
    "uranium_fuel_rod": {
      "name": "Uranium Fuel Rod",
      "labId": "uranium-fuel-rod",
      "stack": 500,
      "iconPos": "-640px -704px"
    },
    "uranium_waste": {
      "name": "Uranium Waste",
      "labId": "uranium-waste",
      "stack": 500,
      "iconPos": "-704px -704px"
    },
    "versatile_framework": {
      "name": "Versatile Framework",
      "labId": "versatile-framework",
      "stack": 50,
      "iconPos": "-0px -768px"
    },
    "water": {
      "name": "Water",
      "iconPos": "-128px -768px"
    },
    "water_extractor": {
      "name": "Water Extractor",
      "labId": "water-extractor",
      "iconPos": "-64px -768px"
    },
    "wire": {
      "name": "Wire",
      "stack": 500,
      "iconPos": "-192px -768px"
    },
    "wood": {
      "name": "Wood",
      "stack": 200,
      "iconPos": "-256px -768px"
    },
    "yellow_power_slug": {
      "name": "Yellow Power Slug",
      "labId": "yellow-power-slug",
      "stack": 50,
      "iconPos": "-320px -768px"
    }
  }
}