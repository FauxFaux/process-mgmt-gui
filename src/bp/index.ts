import type { Blueprint } from './blueprints';

// c.f. https://github.com/FauxFaux/factorio-loader/blob/aca1e2a28c69e0ac073391b6498f897d6803d02a/web/pages/recipes.tsx#L247
// c.f. https://github.com/FauxFaux/factorio-loader/blob/aca1e2a28c69e0ac073391b6498f897d6803d02a/web/pages/recipes.tsx#L299
export const mallAssembler = (
  recipeId: string,
  itemId: string,
  itemHumanName: string | undefined,
  inputs: Record<string, number>,
): Blueprint => ({
  entities: [
    {
      entity_number: 1,
      name: 'assembling-machine-2',
      position: {
        x: -124.5,
        y: 130.5,
      },
      recipe: recipeId,
    },
    {
      control_behavior: {
        connect_to_logistic_network: true,
        logistic_condition: {
          comparator: '<',
          constant: 5,
          first_signal: {
            name: itemId,
            type: 'item',
          },
        },
      },
      direction: 6,
      entity_number: 2,
      name: 'fast-inserter',
      position: {
        x: -122.5,
        y: 130.5,
      },
    },
    {
      direction: 2,
      entity_number: 3,
      name: 'fast-inserter',
      position: {
        x: -122.5,
        y: 131.5,
      },
    },
    {
      entity_number: 4,
      name: 'logistic-chest-passive-provider',
      position: {
        x: -121.5,
        y: 130.5,
      },
    },
    {
      entity_number: 5,
      name: 'logistic-chest-requester',
      position: {
        x: -121.5,
        y: 131.5,
      },
      request_filters: Object.entries(inputs).map(([name, count], i) => ({
        count,
        index: i + 1,
        name,
      })),
    },
  ],
  icons: [
    {
      index: 1,
      signal: {
        name: 'assembling-machine-2',
        type: 'item',
      },
    },
    {
      index: 2,
      signal: {
        name: itemId,
        type: 'item',
      },
    },
  ],
  item: 'blueprint',
  label: 'mall' + (itemHumanName ? `: ${itemHumanName}` : ''),
  version: 281479277838336,
});
