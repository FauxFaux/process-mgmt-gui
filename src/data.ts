import type { ModifierStyle } from './modifiers';
import type { Data } from 'process-mgmt/dist/structures.js';

export interface Lab {
  items: Record<string, LabItem>;
  processes: Record<string, LabProcess>;
}

export interface LabItem {
  // undefined: same as our id, null: synthetic
  labId?: string | null;
  name: string;
  iconPos?: string;
  contained?: true;
  stack?: number;
}

export interface LabProcess {
  name: string;
  iconPos?: string;
  contained?: true;
}

const ds = (name: string, duration: ModifierStyle, output: ModifierStyle) =>
  [name, duration, output] as const;

export const dataSets = {
  'for-the-crown-3.8.3': ds('For The Crown (3.8.3)', 'raw', 'raw'),
  dsp: ds('DSP', 'raw', 'raw'),
  'factorio-ab-1.1.38': ds('Factorio AB (1.1.38)', 'additional', 'additional'),
  'factorio-py-1.1.53': ds('Factorio PY (1.1.53)', 'additional', 'additional'),
  'factorio-ff-1.1.76': ds('Factorio FF (1.1.76)', 'additional', 'additional'),
  'factorio-ff-1.1.94': ds('Factorio FF (1.1.94)', 'additional', 'additional'),
  'factorio-1.1.104-very-bz-0.5.1': ds(
    'Factorio 1.1.104, Very BZ 0.5.1',
    'additional',
    'additional',
  ),
  'plan-b-terraform': ds('Plan B, Terraform', 'raw', 'raw'),
  satisfactory: ds('Satisfactory', 'normal', 'normal'),
  vt: ds('Voxel Tycoon', 'raw', 'raw'),
};

export const toLab: Record<DataSetId, string | null> = {
  'for-the-crown-3.8.3': null,
  dsp: 'dsp',
  'factorio-ab-1.1.38': 'bobang',
  'factorio-py-1.1.53': 'pysalf',
  'factorio-ff-1.1.76': 'ffw',
  'factorio-ff-1.1.94': 'ffw',
  'factorio-1.1.104-very-bz-0.5.1': 'vbz',
  'plan-b-terraform': null,
  satisfactory: 'sfy',
  vt: null,
};

export interface DataSet {
  pm: Data;
  lab?: Lab;
  ico?: string;
}

export type DataSetId = keyof typeof dataSets;

export const loadProcMgmt = async (id: DataSetId): Promise<Data> =>
  (await import(`process-mgmt/dist/${id}/data.js`)).default;

export const loadDataSet = async (id: DataSetId): Promise<DataSet> => {
  const pm = await loadProcMgmt(id);
  const labId = toLab[id];

  let lab, ico;
  if (labId) {
    [lab, ico] = await Promise.all([
      import(`../data/${labId}.json`).then((m) => m.default),
      import(`../data/${labId}.webp`).then((m) => m.default),
    ]);

    preloadImage(ico);
  }

  return { pm, lab, ico };
};

const preloadImage = (src: string) => {
  if (typeof Image === 'undefined') return;
  const img = new Image();
  img.src = src;
};
