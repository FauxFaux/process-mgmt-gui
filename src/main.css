.icon-sprite {
    width: 64px;
    height: 64px;
    scale: 0.4;
    margin: -18px;
    border: 0;
    display: inline-block;
    vertical-align: middle;
}

.item--raw {
    font-family: monospace;
}