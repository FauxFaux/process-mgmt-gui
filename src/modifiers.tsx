export type ModifierStyle = 'raw' | 'normal' | 'additional';
export type DirectMultiplier = number;

export interface Modifier {
  mode: ModifierStyle;
  amount: DirectMultiplier;
}
