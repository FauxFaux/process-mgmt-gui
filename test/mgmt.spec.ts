import { describe, expect, it } from '@jest/globals';

import {
  applyHints,
  computeUnknowns,
  makeInputs,
  updateInputsWithHints,
} from '../src/backend/mgmt';
import { loadDataSet } from '../src/data';
import type { Line } from '../src/components/requirement-table';

describe('mgmt backend', () => {
  it('boots', async () => {
    const ds = await loadDataSet('factorio-ff-1.1.94');
    let reqs: Line[] = [
      { item: 'transport-belt', req: { op: 'produce', amount: 1 } },
    ];
    const inputs = makeInputs(ds, reqs, [
      {
        id: 'transport-belt',
        durationModifier: { mode: 'raw', amount: 1 },
        outputModifier: { mode: 'raw', amount: 1 },
      },
    ]);

    const unknowns = computeUnknowns(inputs);
    expect(unknowns).toMatchSnapshot();

    reqs = applyHints(reqs, unknowns);
    expect(reqs).toMatchSnapshot();

    updateInputsWithHints(inputs, reqs, unknowns);
    expect({
      imports: inputs.imports,
      exports: inputs.exports,
    }).toMatchSnapshot();
  });
});
