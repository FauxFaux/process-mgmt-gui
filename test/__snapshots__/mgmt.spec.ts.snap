// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mgmt backend boots 1`] = `
{
  "iron-gear-wheel": "import",
  "iron-plate": "import",
  "transport-belt": "export",
}
`;

exports[`mgmt backend boots 2`] = `
[
  {
    "item": "transport-belt",
    "req": {
      "amount": 1,
      "hint": "export",
      "op": "produce",
    },
  },
  {
    "item": "iron-gear-wheel",
    "req": {
      "amount": 1,
      "hint": "import",
      "op": "auto",
    },
  },
  {
    "item": "iron-plate",
    "req": {
      "amount": 1,
      "hint": "import",
      "op": "auto",
    },
  },
]
`;

exports[`mgmt backend boots 3`] = `
{
  "exports": [],
  "imports": [
    "iron-gear-wheel",
    "iron-plate",
  ],
}
`;
